File: desafio-03/ronaldofas/csharp/palindromos/Palindromo.cs
using System;

namespace palindromos;

public class Palindromo
{
    public Palindromo() { }

    public bool EhPalindromo(ulong numero){
        // Converte o número para uma string.
        string numeroStr = numero.ToString();

        // Inverte a string.
        char[] charArray = numeroStr.ToCharArray();
        Array.Reverse(charArray);
        string numeroInvertidoStr = new string(charArray);

        // Compara a string original com a string invertida.
        return numeroStr.Equals(numeroInvertidoStr);
    }

    public List<ulong> PalindromosEntre(string inicio, string fim)
    {
        ValidarParametros(inicio);
        ValidarParametros(fim);

        List<ulong> resultado = new List<ulong>();

        for (ulong i = UInt64.Parse(inicio); i <= UInt64.Parse(fim); i++)
        {
            if (EhPalindromo(i))
            {
                resultado.Add(i);
            }
        }

        return resultado;
    }

    private static void ValidarParametros(string numero)
    {
        try
        {
            var conversao = UInt64.Parse(numero);
        } catch (OverflowException)
        {
            throw new OverflowException("Valor maior ou menor que o permitido");
        } catch (ArgumentNullException)
        {
            throw new ArgumentNullException("Valor não pode ser nulo");
        } catch (FormatException)
        {
            throw new FormatException("Valor inválido");
        }
    }
}

File: desafio-03/KarineFBrito/java/Palindromo.java
import java.util.Scanner;

public class Palindromo {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    System.out.println("Escolha o número inicial: ");
    int n1 = sc.nextInt();
    System.out.println("Escolha o numero final: ");
    int n2 = sc.nextInt();
    if (n1 < 0 || n2 < 0) {
      System.out.println("Os números não podem ser negativos!");
      sc.close();
      System.exit(0);
    }
    if (n1 >= n2) {
      System.out.println("O número inicial não pode ser maior ou igual que o final!");
      sc.close();
      System.exit(0);
    }
    for (int i = n1; i <= n2; i++) {
      if (ehPalindromo(i)) {
        System.out.print(i + " ");
      }
    }
  }

  public static boolean ehPalindromo(int numero) {
    if (numero >= 0 && numero < 10) {
      return true;
    }
    String numeroStr = String.valueOf(numero);
    StringBuilder sb = new StringBuilder(numeroStr).reverse();
    return numeroStr.equals(sb.toString());
  }
}
